executor/experiment_configmap.yml

```diff
@@ -1,56 +1,139 @@
-  shell: >
-    kubectl get chaosexperiment -n {{ c_app_ns }} -o jsonpath='{.items[?(@.metadata.name=="{{ c_experiment_name }}")].spec.definition.configmaps[0].name}'
-  args:
-    executable: /bin/bash
-  register: c_map_name
+  - name: Fetching name of configmap
+    shell: >
+      kubectl get chaosexperiment -n {{ c_app_ns }} -o jsonpath='{.items[?(@.metadata.name=="{{ c_experiment_name }}")].spec.definition.configmaps[0].name}'
+    args:
+      executable: /bin/bash
+    register: c_map_name
 
-- name: Creating configmap
-  shell:
-    kubectl create configmap {{c_map_name.stdout}} --from-file=parameters.yml -n {{c_app_ns}}
-  args:
-    executable: /bin/bash
+  - name: Creating configmap
+    shell:
+      kubectl create configmap {{c_map_name.stdout}} --from-file=parameters.yml -n {{c_app_ns}}
+    args:
+      executable: /bin/bash
+
+  when: configMap_available != ''
+  
+- block:
+
+  - name: Fetching name of secret
+    shell: >
+      kubectl get chaosexperiment -n {{ c_app_ns }} -o jsonpath='{.items[?(@.metadata.name=="{{ c_experiment_name }}")].spec.definition.secrets[0].name}'
+    args:
+      executable: /bin/bash
+    register: c_secret_name
+
+  - name: Fetching mount path for secret
+    shell: >
+      kubectl get chaosexperiment -n {{ c_app_ns }} -o jsonpath='{.items[?(@.metadata.name=="{{ c_experiment_name }}")].spec.definition.secrets[0].mountPath}'
+    args:
+      executable: /bin/bash
+    register: c_mount_path
+  
+  when: secret_available != ''
 
-- name: Run the chaos experiment job
+- name: Get the job yaml
   shell:
     kubectl run {{ c_experiment_name }}-{{random_string.stdout}} --restart=OnFailure --image={{c_image.stdout}}
     --namespace={{c_app_ns}} --serviceaccount={{ c_svc_acc }} --image-pull-policy=Always 
     --labels={{c_job_labels.stdout}} {{c_env_list}} --dry-run -oyaml --command -- /bin/bash {{c_job_args.stdout}} > cjob.yml
   args:
     executable: /bin/bash
 
-- name: Include the volumeMounts in jobYml
-  lineinfile:
-    dest: cjob.yml
-    insertafter: "resources: {}"
-    state: present
-    line: '        {{item}}'
-  with_items:
-  - "  mountPath: /mnt/"
-  - "- name: parameters"
-  - "volumeMounts:"
-
-- name: Include the volumes in jobYml
-  lineinfile:
-    dest: cjob.yml
-    insertafter: "serviceAccountName"
-    state: present
-    line: '      {{item}}'
-  with_items:
-  - "    name: {{c_map_name.stdout}}"
-  - "  configMap:"
-  - "- name: parameters"
-  - "volumes:"
+- block:
+
+  - name: Include the volumeMounts in jobYml - configmap only
+    lineinfile:
+      dest: cjob.yml
+      insertafter: "resources: {}"
+      state: present
+      line: '        {{item}}'
+    with_items:
+    - "  mountPath: /mnt/"
+    - "- name: parameters"
+    - "volumeMounts:"
+
+  - name: Include the volumes in jobYml  - configmap only
+    lineinfile:
+      dest: cjob.yml
+      insertafter: "serviceAccountName"
+      state: present
+      line: '      {{item}}'
```

Comment:
Here also, extra space

Reply:
Is this space required?

Reply:
This space is required to maintain the indentation